publisher/managers.py

```diff
@@ -16,7 +18,12 @@ def drafts(self):
 
     def published(self):
         from .models import PublisherModelBase
-        return self.filter(publisher_is_draft=PublisherModelBase.STATE_PUBLISHED)
+
+        return self.filter(
+            Q(publication_start_date__isnull=True) | Q(publication_start_date__lte=timezone.now()),
+            Q(publication_end_date__isnull=True) | Q(publication_end_date__gt=timezone.now()),
+            publisher_is_draft=PublisherModelBase.STATE_PUBLISHED,
+        )
```

Comment:
Hmm.. Should the functionality of this manager method match with the is_published method in PublisherModelBase? I think it should... so maybe add `visible` method rather than changing this one?

Reply:
Yes and no... Don't know ;)

The idea was not to change the current API. Existing software that uses published() also considers start/end date.

But yes, it's a little messy. What should I do?

Reply:
Yeah, I see. Maybe the `is_published` property should follow the same logic then? If there is a need for another property which doesn't consider the date, it should have a different name. How about `is_publishable`? (Yep, naming is hard...)

Reply:
Yes, naming... Hm...

Maybe we should just risk to ignore start/end date and "refactor" the names a little bit:

instance bool properties:

- instance.is_draft
- instance.is_published
- instance.hidden_by_end_date
- instance.hidden_by_start_date
- instance.is_visible
- instance.is_dirty

Model Manager methods:

- PublisherManager.drafts()
- PublisherManager.published()
- PublisherManager.visible()

QuerySet methods:

- PublisherQuerySet.filter_drafts()
- PublisherQuerySet.filter_published()
- PublisherQuerySet.filter_visible()

It should be documented that user should switch from `PublisherManager.published()` to `PublisherManager.visible()`

Reply:
I think that is a good call.

Just a one thing: The QuerySet methods should have same names as the Model Manager methods. See https://docs.djangoproject.com/en/1.8/topics/db/managers/#from-queryset for instructions how to define the `drafts`, `published and `visible` in the query set so that they are also defined in the manager.

Reply:
I have done this in #14 ... It is time-consuming to maintain all changes in separate pull requests. So I created a new pull request with all changes.